src/main/scala/org/camunda/feel/impl/builtin/ConversionBuiltinFunctions.scala

```diff
@@ -101,6 +65,8 @@ object ConversionBuiltinFunctions {
         ValLocalDateTime(dateTime.toLocalDate().atTime(time))
       case List(ValDateTime(dateTime), ValTime(time)) =>
         ValDateTime(time.withDate(dateTime.toLocalDate()))
+      case List(ValDateTime(date), ValString(timezone)) =>
+        ValDateTime(date.withZoneSameInstant(ZoneId.of(timezone)))
```

Comment:
:+1: Looks good.

:x: We should also support date-time values without timezone (i.e. `ValLocalDateTime`). So, it transforms a local date-time into a date-time with a timezone.